packages/story-editor/src/components/panels/design/textStyle/stylePresets/stylePresets.js

```diff
@@ -0,0 +1,188 @@
+import { __ } from '@web-stories-wp/i18n';
+import styled from 'styled-components';
+import { useMemo, useRef, useState } from '@web-stories-wp/react';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../../../app';
+import { PRESET_TYPES } from '../../../../../constants';
+import useAddPreset from '../../../../../utils/useAddPreset';
+import { focusStyle } from '../../../shared';
+import Popup from '../../../../popup';
+import useInspector from '../../../../inspector/useInspector';
+import useApplyStyle from './useApplyStyle';
+import StyleGroup from './styleGroup';
+import StyleManager from './styleManager';
+
+const PresetsHeader = styled.div`
+  display: flex;
+  padding: 8px 0;
+  justify-content: space-between;
+`;
+
+const StylesWrapper = styled.div``;
+
+const SubHeading = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.secondary};
+  margin: 6px 0;
+`;
+
+const StyledButton = styled(Button)`
+  margin-left: auto;
+`;
+
+const StyledMoreButton = styled(Button)`
+  ${focusStyle};
+  margin: 12px 0;
+  padding: 0 16px;
+  justify-content: center;
+  align-self: center;
+  svg {
+    transform: rotate(-90deg);
+    height: 32px;
+    width: 32px;
+    margin-left: -4px;
+  }
+`;
+
+const NoStylesWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: calc(100% + 32px);
+  margin-left: -16px;
+  background-color: ${({ theme }) => theme.colors.opacity.black24};
+  height: 64px;
+`;
+
+const NoStylesText = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+
+function PresetPanel({ pushUpdate }) {
+  const { globalStoryStyles } = useStory(
+    ({
+      state: {
+        story: { globalStoryStyles },
+      },
+    }) => {
+      return {
+        globalStoryStyles,
+      };
+    }
+  );
+
+  const {
+    refs: { inspector },
+  } = useInspector();
+  const buttonRef = useRef(null);
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+
+  const { textStyles } = globalStoryStyles;
+  const hasPresets = textStyles.length > 0;
+
+  const handleApplyStyle = useApplyStyle({ pushUpdate });
+  const { addGlobalPreset } = useAddPreset({ presetType: PRESET_TYPES.STYLE });
+
+  const handlePresetClick = (preset) => {
+    handleApplyStyle(preset);
+  };
+
+  const spacing = useMemo(() => ({ x: 20 }), []);
+  return (
+    <>
+      <PresetsHeader>
+        <SubHeading size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {__('Recent Saved Styles', 'web-stories')}
+        </SubHeading>
+        <StyledButton
+          type={BUTTON_TYPES.TERTIARY}
+          size={BUTTON_SIZES.SMALL}
+          variant={BUTTON_VARIANTS.SQUARE}
+          onClick={addGlobalPreset}
+          aria-label={__('Add style', 'web-stories')}
+        >
+          <Icons.Plus />
+        </StyledButton>
+      </PresetsHeader>
+      {hasPresets && (
+        <>
+          <StylesWrapper>
+            <StyleGroup
+              styles={textStyles.slice(-2)}
```

Comment:
I don't feel like this ordering works well. I think we should reverse the order of the text styles displayed in both lists (both panel and flyout) and then just slice the top 2 here instead:

![Screen Shot 2021-11-08 at 15 48 10](https://user-images.githubusercontent.com/637548/140817713-c0f9ca52-e851-4a0c-809b-66c390bc25e9.png)

Currently, the two displayed are the bottom two in the flyout, but it seems a bit counterintuitive to spot that if the total number is odd.

If we reverse the list, it will effectively be "newest first" which I feel makes more sense.